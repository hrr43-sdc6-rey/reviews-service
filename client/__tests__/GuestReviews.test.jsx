import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16.1';
import GuestReviews from '../components/GuestReviews.jsx';

Enzyme.configure({ adapter: new Adapter() });

describe('GuestReviews', () => {
  it('GuestReviews should render with 0 overAll stars passed in', () => {
    const overallStar = 0;
    const wrapper = shallow(<GuestReviews overallStar={overallStar} />);
    expect(wrapper).toBeDefined();
  });

  it('GuestReviews should render with 1 overAll star passed in', () => {
    const overallStar = 1;
    const wrapper = shallow(<GuestReviews overallStar={overallStar} />);
    expect(wrapper).toBeDefined();
  });

  it('GuestReviews render of 2 stars shoud mach the snapshot', () => {
    const overallStar = 2;
    const wrapper = shallow(<GuestReviews overallStar={overallStar} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('GuestReviews render of 3 stars shoud mach the snapshot', () => {
    const overallStar = 3;
    const wrapper = shallow(<GuestReviews overallStar={overallStar} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('GuestReviews render of 4 stars shoud mach the snapshot', () => {
    const overallStar = 4;
    const wrapper = shallow(<GuestReviews overallStar={overallStar} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('GuestReviews render of 5 stars shoud mach the snapshot', () => {
    const overallStar = 5;
    const wrapper = shallow(<GuestReviews overallStar={overallStar} />);
    expect(wrapper).toMatchSnapshot();
  });
});
