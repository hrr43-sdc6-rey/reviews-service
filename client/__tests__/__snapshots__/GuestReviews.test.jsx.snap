// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuestReviews GuestReviews render of 2 stars shoud mach the snapshot 1`] = `
<div
  className="guestReviews"
>
  <div
    className="aggRating"
  >
    <h2>
      Guest reviews
    </h2>
    <div
      className="starsNum"
    >
      <div
        className="aggNum"
      >
        2
      </div>
      <div
        className="aggStars"
      >
        ★★☆☆☆
      </div>
    </div>
  </div>
</div>
`;

exports[`GuestReviews GuestReviews render of 3 stars shoud mach the snapshot 1`] = `
<div
  className="guestReviews"
>
  <div
    className="aggRating"
  >
    <h2>
      Guest reviews
    </h2>
    <div
      className="starsNum"
    >
      <div
        className="aggNum"
      >
        3
      </div>
      <div
        className="aggStars"
      >
        ★★★☆☆
      </div>
    </div>
  </div>
</div>
`;

exports[`GuestReviews GuestReviews render of 4 stars shoud mach the snapshot 1`] = `
<div
  className="guestReviews"
>
  <div
    className="aggRating"
  >
    <h2>
      Guest reviews
    </h2>
    <div
      className="starsNum"
    >
      <div
        className="aggNum"
      >
        4
      </div>
      <div
        className="aggStars"
      >
        ★★★★☆
      </div>
    </div>
  </div>
</div>
`;

exports[`GuestReviews GuestReviews render of 5 stars shoud mach the snapshot 1`] = `
<div
  className="guestReviews"
>
  <div
    className="aggRating"
  >
    <h2>
      Guest reviews
    </h2>
    <div
      className="starsNum"
    >
      <div
        className="aggNum"
      >
        5
      </div>
      <div
        className="aggStars"
      >
        ★★★★★
      </div>
    </div>
  </div>
</div>
`;
